demo/schemas/user-collection.js

```diff
@@ -7,6 +7,17 @@ module.exports = {
     description: 'A collection of users.',
     type: 'object',
     properties: {
+        limit: {
```

Comment:
By being in the schema, it is by definition a resource property. But our route definition is saying "these properties can be sent in as query string parameters in GET requests, so document them and validate them as such."

It would be nice to *require* the query params in the route in order to access them in the callback function to encourage proper documentation. But again this couples the request with the route. I need to find a clean way to solve this, because it'd be very nice to have.